tests/test_urldispatch.py

```diff
@@ -774,7 +774,23 @@ async def test_dynamic_match_unquoted_path(router: web.UrlDispatcher) -> None:
     resource_id = "my%2Fpath%7Cwith%21some%25strange%24characters"
     req = make_mocked_request("GET", f"/path/{resource_id}")
     match_info = await router.resolve(req)
-    assert match_info == {"path": "path", "subpath": unquote(resource_id)}
```

Comment:
This was a deliberate decision initially, so the user gets the unquoted path in the match info.
i.e. We only need to worry about path separators in the routing. Unless they change the default regex, it'd be impossible for a user to get an unquoted `/` in a variable (and I don't think we should worry about the difference even if they do change the regex).